bool is_found
int32 dx
int32 dy
float32 dx_m
float32 dy_m